app/models/upload_form.rb

```diff
@@ -4,26 +4,65 @@ class UploadForm
 
   validate :validate_reading_csv
   validate :validate_csv
+  validate :validate_data_present
+  validate :validate_header
+
+  HEADER_COLS = ["Address line 1", "Address line 2", "Address line 3", "City", "County", "Postcode", "IP address"].freeze
 
   def validate_reading_csv
-    @data ||= CSV.read(upload_file, col_sep: "\t")
+    @data ||= CSV.read(upload_file)
```

Comment:
I just read up a bit on `CSV` on the ruby docs, and now we're using headers you could use `headers: true` here, which then returns a `CSV::Table` object which may make some of the code a bit simpler. 

There are a few more options described here: https://ruby-doc.org/stdlib-3.0.0/libdoc/csv/rdoc/CSV.html which may be useful? What do you think